plugin/backend_linux.go

```diff
@@ -353,6 +401,64 @@ func (pm *Manager) CreateFromContext(ctx context.Context, tarCtx io.Reader, opti
 	return nil
 }
 
+func splitConfigRootFSFromTar(in io.ReadCloser, config *[]byte) io.ReadCloser {
+	pr, pw := io.Pipe()
+	go func() {
+		tarReader := tar.NewReader(in)
+		tarWriter := tar.NewWriter(pw)
+		defer in.Close()
+
+		hasRootfs := false
+
+		for {
+			hdr, err := tarReader.Next()
+			if err == io.EOF {
+				if !hasRootfs {
```

Comment:
Config can be easily checked from the caller side in https://github.com/docker/docker/pull/29487/files#diff-28e4bbe4b8a4c1b80bf43d15a145427dR557 where it is cleaner. No way to check rootfs from the caller side so it needs to be included in the middle of tar creation.